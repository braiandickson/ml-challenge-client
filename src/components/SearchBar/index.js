import React, { useState } from 'react';
import cn from 'classnames';

import searchIcon from './assets/searchIcon.svg';

import styles from './styles.module.scss';

const SearchBar = ({ handleSearch }) => {
  const [query, setQuery] = useState('');

  const handleChange = (event) => {
    const { value } = event.target;
    setQuery(value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    if (query && !!query.length) {
      handleSearch(query);
    }
  };

  return (
    <form
      className={cn('row middle full-width', styles.container)}
      onSubmit={handleSubmit}
    >
      <input
        name="query"
        type="text"
        className={styles.input}
        onChange={handleChange}
      />
      <button type="submit" className={styles.button}>
        <img className={styles.icon} src={searchIcon} alt="search-icon" />
      </button>
    </form>
  );
};

export default SearchBar;
