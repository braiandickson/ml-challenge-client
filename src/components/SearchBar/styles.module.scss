@import '../../styles/variables/colors';

.container {
  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.2);
}

input {
  border-radius: 2px 0 0 2px;
  font-size: 16px;
  height: 40px;
  padding: 8px 15px;
  width: 550px;

  &:focus {
    outline: none;
  }
}

.button {
  border-radius: 0 2px 2px 0;
  border-left: 1px solid $grey;
  background-color: $white;
  cursor: pointer;
  height: 40px;
  padding: 5px;
  width: 46px;

  &:focus {
    outline: none;
  }
}

.icon {
  height: 18px;
  width: 18px;
}
