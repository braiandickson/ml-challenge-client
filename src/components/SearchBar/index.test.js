import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import SearchBar from './index';

describe('#SearchBar', () => {
  const handleSearch = jest.fn();
  it('Mounts the SearchBar component correctly with data', () => {
    const component = mount(<SearchBar handleSearch={handleSearch} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
