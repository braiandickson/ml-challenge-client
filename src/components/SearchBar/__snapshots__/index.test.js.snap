// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#SearchBar Mounts the SearchBar component correctly with data 1`] = `
<SearchBar
  handleSearch={[MockFunction]}
>
  <form
    className="row middle full-width searchbar"
    onSubmit={[Function]}
  >
    <input
      className="searchbar__input"
      name="query"
      onChange={[Function]}
      type="text"
    />
    <button
      className="searchbar__button"
      type="submit"
    >
      <img
        alt="search-icon"
        className="searchbar__icon"
        src="searchIcon.svg"
      />
    </button>
  </form>
</SearchBar>
`;
